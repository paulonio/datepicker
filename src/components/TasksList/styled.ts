import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
`;

export const Title = styled.h3`
  color: #333333;
  font-size: 14px;
  font-weight: 700;
  line-height: 19px;
`;

export const TaskElement = styled.div`
  color: #333333;
  font-size: 13px;
  font-weight: 400;
  line-height: 17px;
`;
