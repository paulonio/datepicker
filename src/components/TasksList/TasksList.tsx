import React, { FC } from 'react';
import { TaskElement, Title, Wrapper } from './styled';
import calendar from '@utils/ClassCalendar';
import type { Task } from '@components/Modal/Modal';

interface TasksListProps {
  tasks: Task[];
  date: Date;
}

const TasksList: FC<TasksListProps> = ({ tasks, date }) => {
  const dateString = calendar.parseDateToString(date);

  return (
    <Wrapper>
      <Title data-testid="title">Tasks - {dateString}</Title>
      {tasks.length &&
        tasks.map(({ id, title }) => (
          <TaskElement key={id} data-testid="task">
            {title}
          </TaskElement>
        ))}
    </Wrapper>
  );
};

export default TasksList;
