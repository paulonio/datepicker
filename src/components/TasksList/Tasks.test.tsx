import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import TasksList from '@components/TasksList/TasksList';
import type { Task } from '@components/Modal/Modal';

describe('TasksList', () => {
  const tasksData: Task[] = [
    { id: '1', title: 'Learn react' },
    { id: '2', title: 'Learn redux' },
  ];
  const emptyTasksData: Task[] = [];
  const date = new Date(2023, 4, 11);

  it('should render Tasks component', () => {
    const { container } = render(<TasksList tasks={emptyTasksData} date={date} />);

    expect(container).toMatchSnapshot();
  });

  it('should display correct title', () => {
    render(<TasksList tasks={emptyTasksData} date={date} />);

    const title = screen.getByTestId('title');
    expect(title).toHaveTextContent('Tasks - 11/05/2023');
  });

  it('should not display tasks on empty data', () => {
    render(<TasksList tasks={emptyTasksData} date={date} />);

    const task = screen.queryAllByTestId('task');
    expect(task).toHaveLength(0);
  });

  it('should display tasks on correct data', () => {
    render(<TasksList tasks={tasksData} date={date} />);

    const tasks = screen.getAllByTestId('task');
    expect(tasks).toHaveLength(2);
    expect(tasks[0]).toHaveTextContent('Learn react');
  });
});
