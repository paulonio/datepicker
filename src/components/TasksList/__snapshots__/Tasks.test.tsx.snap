// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksList should render Tasks component 1`] = `
<div>
  <div
    class="sc-beqWaB iFODHc"
  >
    <h3
      class="sc-gueYoa ckhcaG"
      data-testid="title"
    >
      Tasks - 
      11/05/2023
    </h3>
  </div>
</div>
`;
